ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -5868,7 +5878,11 @@ static int MMWriteVectorMetadataFile(struct MiraMonVectLayerInfo *hMiraMonLayer,
                    sizeof(hMMMD.hBB));
             hMMMD.pLayerDB = nullptr;
         }
-        return MMWriteMetadataFile(&hMMMD);
+        hMMMD.aArcFile = strdup_function(
+            get_filename_function(hMiraMonLayer->MMPolygon.pszLayerName));
+        nResult = MMWriteMetadataFile(&hMMMD);
+        free_function(hMMMD.aArcFile);
```

Comment:
shouldn't hMMMD.aArcFile be set to NULL after free_function, to avoid a potential reuse of freed memory ?

Reply:
I think it's not necessary because it's in a structure created at the same function and set to 0.
memset(&hMMMD, 0, sizeof(hMMMD));

And after the use the function finalizes with return 0 and this structure is not used outside.

Reply:
ok, I missed this was a local structure